sonar-db/src/main/java/org/sonar/db/version/v54/MigrateDisabledUsersToOnlyKeepLoginAndName.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.db.version.v54;
+
+import java.sql.SQLException;
+import org.sonar.api.utils.System2;
+import org.sonar.db.Database;
+import org.sonar.db.version.BaseDataChange;
+import org.sonar.db.version.MassUpdate;
+import org.sonar.db.version.Select;
+import org.sonar.db.version.SqlStatement;
+
+/**
+ * Update all disabled users to only kept login and name
+ */
+public class MigrateDisabledUsersToOnlyKeepLoginAndName extends BaseDataChange {
+
+  private final System2 system2;
+
+  public MigrateDisabledUsersToOnlyKeepLoginAndName(Database db, System2 system2) {
+    super(db);
+    this.system2 = system2;
+  }
+
+  @Override
+  public void execute(Context context) throws SQLException {
+    MassUpdate update = context.prepareMassUpdate().rowPluralName("users");
+    update.select("SELECT u.id FROM users u WHERE u.active=? " +
+      "AND email IS NOT NULL " +
+      "AND scm_accounts IS NOT NULL " +
+      "AND external_identity IS NOT NULL " +
+      "AND external_identity_provider IS NOT NULL " +
+      "AND crypted_password IS NOT NULL " +
+      "AND salt IS NOT NULL " +
+      "AND remember_token IS NOT NULL " +
+      "AND remember_token_expires_at IS NOT NULL "
+    ).setBoolean(1, false);
+    update.update("UPDATE users SET email=null, scm_accounts=null, " +
+      "external_identity=null, external_identity_provider=null, " +
+      "crypted_password=null, salt=null, " +
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Remove this hard-coded password. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2068)